src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -165,12 +158,177 @@ private static JToken ParseToJToken(string value)
+                                        historyItem.Remove("Result");
+                                    }
+
+                                    ConvertOutputToJToken(historyItem, showHistoryOutput);
+                                    break;
+                                case EventType.ExecutionTerminated:
+                                    historyItem.Remove("Input");
+                                    break;
+                                case EventType.TimerFired:
+                                    historyItem.Remove("TimerId");
+                                    break;
+                                case EventType.EventRaised:
+                                    historyItem.Remove("Input");
+                                    break;
+                                case EventType.OrchestratorStarted:
+                                case EventType.OrchestratorCompleted:
+                                    indexList.Add(i);
+                                    break;
+                            }
+
+                            historyItem.Remove("EventId");
+                            historyItem.Remove("IsPlayed");
+                        }
+                    }
+
+                    var counter = 0;
+                    indexList.Sort();
+                    foreach (var indexValue in indexList)
+                    {
+                        historyArray.RemoveAt(indexValue - counter);
+                        counter++;
+                    }
+                }
+            }
+
+            return new DurableOrchestrationStatus
+            {
+                Name = orchestrationState.Name,
+                InstanceId = orchestrationState.OrchestrationInstance.InstanceId,
+                CreatedTime = orchestrationState.CreatedTime,
+                LastUpdatedTime = orchestrationState.LastUpdatedTime,
+                RuntimeStatus = (OrchestrationRuntimeStatus)orchestrationState.OrchestrationStatus,
+                Input = ParseToJToken(orchestrationState.Input),
+                Output = ParseToJToken(orchestrationState.Output),
+                History = historyArray,
+            };
+        }
+
+        private static void TrackNameAndScheduledTime(JObject historyItem, EventType eventType, int index, Dictionary<string, EventIndexDateMapping> eventMapper)
+        {
+            eventMapper.Add($"{eventType}_{historyItem["EventId"]}", new EventIndexDateMapping { Index = index, Name = (string)historyItem["Name"], Date = (DateTime)historyItem["Timestamp"] });
+        }
+
+        private static void AddScheduledEventDataAndAggregate(ref Dictionary<string, EventIndexDateMapping> eventMapper, string prefix, JToken historyItem, List<int> indexList)
+        {
+            if (!eventMapper.ContainsKey($"{prefix}_{historyItem["TaskScheduledId"]}"))
+            {
+                return;
+            }
+
+            var taskScheduledData = eventMapper[$"{prefix}_{historyItem["TaskScheduledId"]}"];
```

Comment:
I don't think the null checks you have in this method are necessary. Also, please use `TryGetValue` instead of `ContainsKey` followed by an index lookup. `TryGetValue` is more efficient since you're only querying the dictionary once. Here is an alternate implementation of this method:
```csharp
if (eventMapper.TryGetValue($"{prefix}_{historyItem["TaskScheduledId"]}", out EventIndexDateMapping taskScheduledData))
{
    historyItem["ScheduledTime"] = taskScheduledData.Date;
    historyItem["FunctionName"] = taskScheduledData.Name;
    indexList.Add(taskScheduledData.Index);
}
```

Reply:
Great! I need to use this more to be my first option in similar scenarios. Thank you!